In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('words.csv', index_col='Word')

In [3]:
df.head()

Char Count  Value
Word                     
aa               2      2
aah              3     10
aahed            5     19
aahing           6     40
aahs             4     29

## How many elements does this data frame have?

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 172821 entries, aa to zyzzyvas
Data columns (total 2 columns):
 #   Column      Non-Null Count   Dtype
---  ------      --------------   -----
 0   Char Count  172821 non-null  int64
 1   Value       172821 non-null  int64
dtypes: int64(2)
memory usage: 4.0+ MB


In [5]:
#172821

## What is the value of the word 'microspectrophotometries'?

In [6]:
value_of_microspectrophotometries = df.loc["microspectrophotometries", "Value"]
value_of_microspectrophotometries

317

In [7]:
print(f'Value of microspectrophotometries is {value_of_microspectrophotometries}')

Value of microspectrophotometries is 317


## What is the highest value of a word in dataframe?

In [8]:
highest_value = df["Value"].max()
highest_value

319

In [9]:
print(f'highest value of a word in dataframe is {highest_value}')

highest value of a word in dataframe is 319


## Which of the following words have a Char Count of 7 and a Value of 87 ?
## pinfish, glowing, superheterodyne, enfold, microbrew 

In [10]:
Options = ['pinfish', 'glowing', 'superheterodyne', 'enfold', 'microbrew']

In [11]:
filtered_words = df[(df.index.isin(Options)) & (df['Char Count'] == 7) & (df['Value'] == 87)]

In [12]:
filtered_words

Char Count  Value
Word                      
glowing           7     87

In [13]:
print(f'The answer is {filtered_words.index[0]}')

The answer is glowing


## What is the highest possible length of a word?

In [14]:
longest_word = df['Char Count'].max()
longest_word

28

In [15]:
print(f'highest possible length of a word is {longest_word}')

highest possible length of a word is 28


## The word microspectrophotometries is the only word with a value of 317. Find the only word with a value of 319

In [16]:
value_319 = df[df["Value"] == 319]
value_319

Char Count  Value
Word                                      
reinstitutionalizations          23    319

In [17]:
print(f'The only word with a value of 319 is {value_319.index[0]}')

The only word with a value of 319 is reinstitutionalizations


## There are 1921 words with a value of 100, making it the second most common letter sum. What value is most common?

In [18]:
most_common_value = df["Value"].value_counts().head(1)
most_common_value

Value
93    1965
Name: count, dtype: int64

In [19]:
print(f'The most common value is {most_common_value.index[0]}')

The most common value is 93


## What is the shortest word with value 274?

In [20]:
shortest_word_V274 = df[df["Value"] == 274].sort_values("Char Count", ascending = True).head(1)
shortest_word_V274

Char Count  Value
Word                                   
overprotectivenesses          20    274

In [21]:
print(f'The shortest word with value 274 is {shortest_word_V274.index[0]}')

The shortest word with value 274 is overprotectivenesses



## The "Value Ratio" of a word is defined as the Value of a word divided by its length. That is: ValueRatio(word) = Value(word) / len(word). Create a new column in your DataFrame named Ratio that contains such computation.

In [22]:
df['Ratio'] = df["Value"] / df["Char Count"]
df.head(5)

Char Count  Value     Ratio
Word                               
aa               2      2  1.000000
aah              3     10  3.333333
aahed            5     19  3.800000
aahing           6     40  6.666667
aahs             4     29  7.250000

## What is the maximum value of Ratio?

In [23]:
max_ratio =  df["Ratio"].max()
max_ratio

22.5

In [24]:
print(f'Maximum value of Ratio is {max_ratio}')

Maximum value of Ratio is 22.5


## What word is the one with the highest Ratio?

In [25]:
max_ratio_word = df[df["Ratio"] == df["Ratio"].max()]
max_ratio_word

Char Count  Value  Ratio
Word                          
xu             2     45   22.5

In [26]:
print(f'Word with the highest Ratio is {max_ratio_word.index[0]}')

Word with the highest Ratio is xu


## How many words have a Ratio of 10?

In [27]:
Ratio_of_10 = df[df["Ratio"] == 10].shape
Ratio_of_10

(2604, 3)

In [28]:
print(f'There are {Ratio_of_10[0]} words with ratio 10')

There are 2604 words with ratio 10


## What is the maximum Value of all the words with a Ratio of 10?

In [29]:
max_value_of_R10 = df.loc[df["Ratio"] == 10, "Value"].max()
max_value_of_R10

240

In [30]:
print(f'maximum Value of all the words with a Ratio of 10 is {max_value_of_R10}')

maximum Value of all the words with a Ratio of 10 is 240


## Of those words with a Value of 260, what is the lowest Char Count found?

In [31]:
lowest_Char_Count_V260 = df.loc[df["Value"] == 260, "Char Count"].min()
lowest_Char_Count_V260

17

In [32]:
print(f'lowest Char Count of word with value 260 is {lowest_Char_Count_V260}')

lowest Char Count of word with value 260 is 17


## Which word has a Value of 260 and has the lowest Char Count?

In [33]:
word_lowest_Char_Count_V260 = df.loc[(df["Value"] == 260) & (df["Char Count"] == lowest_Char_Count_V260)]
word_lowest_Char_Count_V260

Char Count  Value      Ratio
Word                                           
hydroxytryptamine          17    260  15.294118

In [34]:
print(f'Lowest char count of word having value is {word_lowest_Char_Count_V260.index[0]}')

Lowest char count of word having value is hydroxytryptamine
